File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/ChargeItem.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class ChargeItem : DomainResource, IResource
{
    public Base.ResourceReference[]? Service { get; set; }
    public string[]? DefinitionUri { get; set; }
    public Base.ResourceReference? Enterer { get; set; }
    public Base.ResourceReference? RequestingOrganization { get; set; }
    public Base.CodeableConcept? ProductCodeableConcept { get; set; }
    public Base.ResourceReference? ProductReference { get; set; }
    public string[]? DefinitionCanonical { get; set; }
    public Base.CodeableConcept[]? Bodysite { get; set; }
    public string? OccurrenceTiming { get; set; }
    public Base.ResourceReference? CostCenter { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public Base.ResourceReference[]? Account { get; set; }
    public Base.CodeableConcept[]? Reason { get; set; }
    public object? Product    
    {
        get
        {
            if (ProductCodeableConcept is not null)
            {
                return ProductCodeableConcept;
            }
    
            if (ProductReference is not null)
            {
                return ProductReference;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.CodeableConcept))
            {
                ProductCodeableConcept = (Base.CodeableConcept)value;return;
            }
    
            if (value?.GetType() == typeof(Base.ResourceReference))
            {
                ProductReference = (Base.ResourceReference)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.ResourceReference[]? SupportingInformation { get; set; }
    public Base.Period? OccurrencePeriod { get; set; }
    public required string Status { get; set; }
    public required Base.CodeableConcept Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference? Context { get; set; }
    public Base.Quantity? Quantity { get; set; }
    public Base.ResourceReference[]? PartOf { get; set; }
    public Base.Money? PriceOverride { get; set; }
    public string? EnteredDate { get; set; }
    public string? OccurrenceDateTime { get; set; }
    public string? OverrideReason { get; set; }
    public Base.ResourceReference? PerformingOrganization { get; set; }
    public required Base.ResourceReference Subject { get; set; }
    public string? FactorOverride { get; set; }
    public object? Occurrence    
    {
        get
        {
            if (OccurrenceTiming is not null)
            {
                return OccurrenceTiming;
            }
    
            if (OccurrencePeriod is not null)
            {
                return OccurrencePeriod;
            }
    
            if (OccurrenceDateTime is not null)
            {
                return OccurrenceDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                OccurrenceTiming = (string)value;return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                OccurrencePeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OccurrenceDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public ChargeItem_Performer[]? Performer { get; set; }

    public class ChargeItem_Performer : BackboneElement
    {
        public required Base.ResourceReference Actor { get; set; }
        public Base.CodeableConcept? Function { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/ChargeItem.py
from typing import Optional, List
from pydantic import *
from ..base import *

class ChargeItem_Performer(BackboneElement):
    actor: Reference
    function: Optional[CodeableConcept] = None

class ChargeItem(DomainResource):
    service: Optional[List[Reference]] = None
    definition_uri: Optional[List[str]] = None
    enterer: Optional[Reference] = None
    requesting_organization: Optional[Reference] = None
    product_codeable_concept: Optional[CodeableConcept] = None
    product_reference: Optional[Reference] = None
    definition_canonical: Optional[List[str]] = None
    bodysite: Optional[List[CodeableConcept]] = None
    occurrence_timing: Optional[Timing] = None
    cost_center: Optional[Reference] = None
    note: Optional[List[Annotation]] = None
    account: Optional[List[Reference]] = None
    reason: Optional[List[CodeableConcept]] = None
    supporting_information: Optional[List[Reference]] = None
    occurrence_period: Optional[Period] = None
    status: str
    code: CodeableConcept
    identifier: Optional[List[Identifier]] = None
    context: Optional[Reference] = None
    quantity: Optional[Quantity] = None
    part_of: Optional[List[Reference]] = None
    price_override: Optional[Money] = None
    entered_date: Optional[str] = None
    occurrence_date_time: Optional[str] = None
    override_reason: Optional[str] = None
    performing_organization: Optional[Reference] = None
    subject: Reference
    factor_override: Optional[float] = None
    performer: Optional[List[ChargeItem_Performer]] = None